### PR TITLE
Assert that user is AWS account

### PR DESCRIPTION
**Pull request by janboll**
Prevents errors like: An error occurred (NoSuchEntity) when calling the DeleteLoginProfile operation: The user with name foobar cannot be found.

### CODE REVIEW

File: reconcile/aws_iam_password_reset.py [hunk 1]
```diff
@@ -8,9 +10,23 @@
 QONTRACT_INTEGRATION = "aws-iam-password-reset"
 
 
+def get_roles(roles: list[dict[str, Any]], org_username: str) -> dict[str, Any]:
+    return [d for d in roles if d["org_username"] == org_username][0]
```
Comment (Piojo) [inline reconcile/aws_iam_password_reset.py:14]: Type mismatch: this is a `list` but you're stating `dict`.

File: reconcile/aws_iam_password_reset.py [hunk 2]
```diff
@@ -8,9 +10,23 @@
 QONTRACT_INTEGRATION = "aws-iam-password-reset"
 
 
+def get_roles(roles: list[dict[str, Any]], org_username: str) -> dict[str, Any]:
```
Comment (Piojo) [inline reconcile/aws_iam_password_reset.py:13]: Try to be generic with the input parameters:
```python
from typing import Mapping, Iterable
....
def get_roles(roles: Iterable[Mapping[str, Any]], ...)
```

File: reconcile/aws_iam_password_reset.py [hunk 3]
```diff
@@ -8,9 +10,23 @@
...
+    return [d for d in roles if d["org_username"] == org_username][0]
+
+
+def account_in_roles(roles: list[dict[str, Any]], aws_account: str) -> bool:
+    for role in roles:
+        if role.get("aws_groups"):
+            for g in role.get("aws_groups"):
```
Comment (Piojo) [inline reconcile/aws_iam_password_reset.py:20]: Shorter:
```suggestion
        for g in role.get("aws_groups") or []:
```
(and re-indent)

**Review by fishi0x01 — APPROVED** [review]
Nice! :+1:

**Pull request merged**